docs/rfcs/1011-non-blocking-invite-transport.md

```diff
@@ -0,0 +1,1238 @@
+                // The submitted step is too advanced
+                // Every step before must have been completed by both peers
+                "status": "step_too_advanced"
+            },
+            {
+                // The submitted step somehow changed during polling
+                "status": "step_mismatch"
+            }
+        ],
+        "nested_types": [
+            {
+                // GreeterStep should be identical to the one in invite_claimer_step.json5
+                "name": "GreeterStep",
+                "discriminant_field": "step",
+                "variants": [
+                    [...]  // This is described in the protocol update section
+                ]
+            },
+            {
+                // ClaimerStep should be identical to the one in invite_claimer_step.json5
+                "name": "ClaimerStep",
+                "discriminant_field": "step",
+                "variants": [
+                  [...]  // This is described in the protocol update section
+                ]
+            }
+        ]
+    }
+]
+```
+
+#### `invite_claimer_step` command
+
+This invited command, along with `invite_greeter_step`, is the main route used to establish a secure channel between both peers.
+
+It is used by the claimer to submit the data for a given step. The step is identified by:
+- the provided greeter
+- the invitation token
+- the type of the submitted step data
+
+For the step data to be accepted, it has to pass the following the checks:
+- the greeter exists
+- the greeter is allowed (i.e. it is part of the greeters)
+- the attempt exists
+- the attempt is active (i.e. joined and not cancelled)
+- the previous steps must have been completed by both peers
+
+> [!NOTE]
+> The state of the invitation doesn't have to be checked since it is already checked by the `/invited` HTTP route.
+
+Once the data is accepted, it either returns:
+- the greeter step data if it is available
+- a `not_ready` status otherwise
+
+In the second case, the author can poll the command with the same parameters until it gets the greeter step data.
+
+If the submitted step differs from the one that has already been registered, a dedicated `step_mismatch` status is returned.
+
+Pseudo-code:
+
+```python
+def invite_claimer_step(
+    token: InvitationToken,
+    greeter: UserID,
+    attempt: AttemptID,
+    step: ClaimerStep
+) ->
+  InvitationNotFound | InvitationCompleted | InvitationCancelled
+  GreeterNotFound | GreeterNotAllowed
+  AttemptNotFound | AttemptNotJoined | AttemptCancelled
+  StepTooAdvanced | PayloadMismatch
+  NotReady | OK[GreeterStep]:
+    # Perform checks
+    [...]
+    # Call `set_claimer_payload`
+    [...]
+    # Return `greeter_step` or `NotReady`
+    [...]
+```
+
+Schema definition:
+
+```json5
+[
+    {
+        "major_versions": [
+            4
+        ],
+        "req": {
+            "cmd": "invite_claimer_step",
+            "fields": [
+                {
+                    "name": "token",
```

Comment:
remove `token` and `greeter`, only rely on `attempt`

Reply:
ccf6d568d950206e8b632bc2469462e7910f790c